### PR TITLE
[BACKLOG-35000] CSRF refactor

### PR DESCRIPTION
**Pull request by dcleao**
I've split the PR into relevant commits.

Note that, in this PR, https://github.com/pentaho/pentaho-platform/pull/4847/commits/6a6255cd3d3f844e9357f601872bbae0b3d17bd8?file-filters%5B%5D=.xml#diff-c005f1bf02f71f1c8aeb619e7c2470635b3bdfda6ed18201c289aaeb07e140e1R127-R146, the security-web artifacts are being added to the Pentaho war assembly. This matches with adding their package namespace to the root Karaf context, in https://github.com/pentaho/pentaho-karaf-assembly/pull/651 and https://github.com/pentaho/pentaho-karaf-ee-assembly/pull/313.

Given the latter modifications to the karaf assemblies, are there any other war/assemblies that the security-web artifacts should be being added to?

Please, review:
@graimundo
/cc @bennychow if you want to review, be my guest!
/cc @jcarneirohv

Depends on https://github.com/pentaho/security-web/pull/4.

### CODE REVIEW

**Comment by graimundo** [comment]
>Given the latter modifications to the karaf assemblies, are there any other war/assemblies that the security-web artifacts should be being added to?

Not sure. What is being exported into the OSGi environment must be available in the main classloader. 
Probably at least PDI assemblies ( [CE](https://github.com/pentaho/pentaho-kettle/tree/master/assemblies/lib) and [EE](https://github.com/pentaho/pdi-assemblies-ee/tree/master/core/lib) ) should be updated to make sure the required jars are included.
Not sure if there are any more places where this would be required but I don't think so.

File: assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-cors.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          http://www.hitachivantara.com/schema/security/web/model/spring http://www.hitachivantara.com/schema/security/web/model/spring/hv-swm-spring.xsd"
+       default-lazy-init="true">
+
+  <!-- region CORS Configurations -->
+
+  <!-- Root/Default CORS configuration -->
+  <bean class="com.hitachivantara.security.web.impl.model.cors.CorsRequestSetConfigurationPojo">
+    <pen:publish as-type="INTERFACES" />
+
+    <!-- There can be only one root CORS configuration -->
```
Comment (graimundo) [inline assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-cors.xml:22]: image

**Comment by dcleao** [comment]
Rebased on master.

**Review by graimundo — APPROVED** [review]

**Comment by dcleao** [comment]
Addressed overall code review comments.

**Pull request merged**